### DISCUSSION TITLE
What is the best rendering approach for a product page that can be edited by a user?

### DISCUSSION THREAD

**Discussion by ladel46** [vercel/next.js#66703] [App Router]
Im working on an e-commerce website using next js 14, and was researching about different rendering method and which one to use for each of my pages, and i got stuck in the case of how to render my product pages, basically from what i understood SSG pages in next are rendered at build time, and if data changes after build the page info will not change, this is clearly wrong and not what i want to go for, so i was wondering if their was a way to revalidate the page on demand for all users or something OR if i should switch to a dynamic page for this scenario in the first place

**Comment by SouthLink** [comment]
So it seems that [ISR](https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration) might be the better choice

1. As an e-commerce site, you may need better performance
> Incremental Static Regeneration (ISR) enables you to use static-generation on a per-page basis, without needing to rebuild the entire site. With ISR, you can retain the benefits of static while scaling to millions of pages.

2. You need to update your page dynamically
You can control your cache freshness by revalidate

And you can use the fallback property to control whether the HTML is generated when the build is made or when requested, so that the build time is not extended by having a large number of product pages
